torch/utils/_import_utils.py

```diff
@@ -29,7 +33,7 @@ def dill_available():
 
 
 @functools.lru_cache
-def import_dill():
+def import_dill() -> Optional["dill"]:
```

Comment:
It looks like I was wrong (or maybe just hopeful?) and this is the wrong way to do this. There doesn't seem to be a way to automatically get a Protocol for a module so we end up either having to be more generic (and return `Optional[ModuleType]`) or define our own protocol for the `dill` module.

Coming up a with a protocol in this situation is a bit much so I'd just use `ModuleType`.

Reply:
Did you run it locally? CI looks green to me.

Nevertheless I've updated the PR to use ModuleType since I'm assuming you had signal that this doesn't work.

Reply:
Yes - ran it locally. The problem is that we've got a global override for `dill` itself (in mypy.ini) so to really understand what's going on you need to use something like "typing" which doesn't have special override handling.